types/chart.js/index.d.ts

```diff
@@ -729,6 +729,7 @@ declare namespace Chart {
         type?: ScaleType | string;
         display?: boolean | string;
         id?: string;
+        labels?: Array<string | string[] | number | number[] | Date | Date[] | Moment | Moment[]>;
```

Comment:
According to the documentation this property can be `string[]`, so why all the other types? Especially the nested arrays.

Reply:
Same as `ChartData.labels`, and these types produce valid labels on the chart. Other issues ([one here](https://github.com/DefinitelyTyped/DefinitelyTyped/issues/38583#issuecomment-535384241)) mention "The chart.js documentation is quite a mess", which I don't fully agree with, but I don't think it accurately describes all behavior of the library.

Reply:
@leapwill As discussed [here](https://github.com/chartjs/Chart.js/issues/7933#issuecomment-714309035), labels' types for scales are different from category axis labels. So `string[]` is the correct type for labels.